src/pages/Profile.js

```diff
@@ -236,189 +236,176 @@ const FenceLink = ({ provider: { key, name } }) => {
-      h(LabeledCheckbox, {
-        checked: profileInfo[key] === 'true',
-        onChange: v => this.assignValue(key, v.toString())
-      }, [span({ style: styles.form.checkboxLabel }, [title])])
-    ])
+    loadProxyGroup()
+  })
 
-    return h(Fragment, [
-      line(
-        textField('firstName', 'First Name', { required: true }),
-        textField('lastName', 'Last Name', { required: true })
-      ),
-      line(
-        textField('title', 'Title')
-      ),
-      line(
-        div([
-          div({ style: styles.form.title }, ['Email']),
-          div({ style: { margin: '1rem' } }, [profileInfo.email])
-        ]),
-        textField('contactEmail', 'Contact Email for Notifications (if different)', { placeholder: profileInfo.email })
-      ),
-      line(
-        textField('institute', 'Institution'),
-        textField('institutionalProgram', 'Institutional Program')
-      ),
-
-      div({ style: styles.form.title }, [
-        span({ style: { marginRight: '0.5rem' } }, ['Proxy Group']),
-        h(InfoBox, [
-          'For more information about proxy groups, see the ',
-          h(Link, {
-            href: 'https://support.terra.bio/hc/en-us/articles/360031023592',
-            ...Utils.newTabLinkProps
-          }, ['user guide.'])
-        ])
-      ]),
-      div({ style: { margin: '1rem' } }, [proxyGroup]),
-
-      sectionTitle('Program Info'),
-
-      div({ style: styles.form.title }, ['Non-Profit Status']),
-      div({ style: { margin: '1rem' } }, [
-        radioButton('nonProfitStatus', 'Profit'),
-        radioButton('nonProfitStatus', 'Non-Profit')
-      ]),
-      line(
-        textField('pi', 'Principal Investigator/Program Lead')
-      ),
-      line(
-        textField('programLocationCity', 'City'),
-        textField('programLocationState', 'State')
-      ),
-      line(
-        textField('programLocationCountry', 'Country')
-      ),
-
-      sectionTitle('Account Notifications'),
-
-      checkbox('notifications/GroupAccessRequestNotification', 'Group Access Requested'),
-      checkbox('notifications/WorkspaceAddedNotification', 'Workspace Access Added'),
-      checkbox('notifications/WorkspaceRemovedNotification', 'Workspace Access Removed'),
-
-      h(ButtonPrimary, {
-        style: { marginTop: '3rem' },
-        onClick: () => this.save(),
-        disabled: !!errors,
-        tooltip: !!errors && 'Please fill out all required fields'
-      }, ['Save Profile'])
-    ])
-  }
 
-  assignValue(key, value) {
-    this.setState({ profileInfo: _.set(key, value, this.state.profileInfo) })
-  }
+  // Helpers
+  const assignValue = _.curry((key, value) => {
+    setProfileInfo(_.set(key, value))
+  })
 
-  save = _.flow(
-    Utils.withBusyState(v => this.setState({ saving: v })),
-    withErrorReporting('Error saving profile')
-  )(async () => {
-    const { profileInfo } = this.state
+  const line = children => div({ style: styles.form.line }, children)
```

Comment:
Why do we not need that spread here anymore?

Reply:
For consistency, I changed this helper from taking an arbitrary number of arguments to taking a standard children array.